### PR TITLE
[CI/Docker/Deployment] [oldstable] Refactory

### PR DESCRIPTION
**Pull request by y0urself**
>**The release actions has not been tested for now.**
> **So I will use this branch for testing it for the next community release.**

**What**:

* Updated Dockerimages:
  * `prod` now builds from `build` instead from debian stable.
  * build only uses `gcc` for now
    * `gcc` is the `PRODUCTION` build(-> GOS), so this should be mandatory and default for us, too
    * removed `clang`
* CI now only tests gcc and installs `clang` in job for scan build
* Use `greenbone/actions/release@v1` for releases (untested)

**What**:

**Why**:

**How**:

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: .github/workflows/container.yml [hunk 1]
```diff
@@ -31,59 +58,25 @@ jobs:
           file: .docker/prod.Dockerfile
 
   trigger-related-projects:
-    needs: images
+    needs: production
     name: "Trigger update docker images in related projects"
     runs-on: ubuntu-latest
     steps:
-      - name: Trigger oldstable openvas-scanner container image build
+      - name: Trigger main openvas-scanner container image build
```
Comment (ArnoStiefvater) [inline .github/workflows/container.yml:65]:
```suggestion
      - name: Trigger oldstable openvas-scanner container image build
```

File: .github/workflows/container.yml [hunk 2]
```diff
@@ -31,59 +58,25 @@ jobs:
...
         run: |
           curl -X POST https://api.github.com/repos/greenbone/openvas-scanner/dispatches \
           -H 'Accept: application/vnd.github.everest-preview+json' \
           -u greenbonebot:${{ secrets.GREENBONE_BOT_TOKEN }} \
-          --data '{"event_type": "update-oldstable-images"}'
-      - name: Trigger oldstable gvmd container image build
+          --data '{"event_type": "update-main-images"}'
```
Comment (ArnoStiefvater) [inline .github/workflows/container.yml:70]:
```suggestion
          --data '{"event_type": "update-oldstable-images"}'
```

File: .github/workflows/container.yml [hunk 3]
```diff
@@ -31,59 +58,25 @@ jobs:
...
+      - name: Trigger main gvmd container image build
         run: |
           curl -X POST https://api.github.com/repos/greenbone/gvmd/dispatches \
           -H 'Accept: application/vnd.github.everest-preview+json' \
           -u greenbonebot:${{ secrets.GREENBONE_BOT_TOKEN }} \
-          --data '{"event_type": "update-oldstable-images"}'
-      - name: Trigger oldstable openvas container image build
+          --data '{"event_type": "update-main-images"}'
+      - name: Trigger main openvas container image build
```
Comment (ArnoStiefvater) [inline .github/workflows/container.yml:77]:
```suggestion
      - name: Trigger oldstable openvas container image build
```

File: .github/workflows/container.yml [hunk 4]
```diff
@@ -31,59 +58,25 @@ jobs:
...
-      - name: Build and push
-        uses: docker/build-push-action@v2
-        with:
-          context: .
-          push: true
-          tags: ${{ steps.container.outputs.image-tags }}
-          file: .docker/build.Dockerfile
-          build-args: |
-            COMPILER=${{matrix.compiler}}
+          --data '{"event_type": "update-main-images"}'
```
Comment (ArnoStiefvater) [inline .github/workflows/container.yml:82]:
```suggestion
          --data '{"event_type": "update-oldstable-images"}'
```

File: .docker/build.Dockerfile [hunk 5]
```diff
@@ -20,6 +20,8 @@ RUN apt-get update && \
     build-essential \
     curl \
     cmake \
+    gcc \
+    g++ \
```
Comment (ArnoStiefvater) [inline .docker/build.Dockerfile:24]: gcc is already in `build-essential` and g++ is the for c++.

Comment (y0urself) [inline .docker/build.Dockerfile:24]:
```suggestion
```

**Review by ArnoStiefvater — APPROVED** [review]

**Pull request merged**